### ISSUE TITLE
Bracketed and parenthesized tuples of the same values are considered equal

### ISSUE THREAD

**Issue by ianthehenry** [janet-lang/janet#1073]
It is currently the case that `(= '(1 2 3) '[1 2 3])`, and they hash to the same value, and I think that that should not be the case.

I'm not actually sure if this is the reason for the following problems, but it is my amateur guess, and I *think* they should be considered distinct regardless. But this is how it came up:

```
Janet 1.26.0-c28df14e macos/aarch64 - '(doc)' for help
repl:1:> (defmacro macro1 [] (def a '(1 2 3)) (def b '[1 2 3]) ~(do ,a ,b))
<function macro1>
repl:2:> (macex '(macro1))
(do (1 2 3) (1 2 3))
repl:3:> (defmacro macro2 [] (def a '(1 2 3)) (def b '[1 2 4]) ~(do ,a ,b))
<function macro2>
repl:4:> (macex '(macro2))
(do (1 2 3) [1 2 4])
```

Written like this the problem is pretty obvious, but this can manifest with weird nonlocal problems when writing macros, where writing a quoted form in one file suddenly breaks a macro written in another file by "changing" the bracketedness of one of its constituents.

```
repl:1:> (def a '(1 2 3))
(1 2 3)
repl:2:> (defmacro macro1 [] (def b '[1 2 3]) ~(do ,a ,b))
<function macro1>
repl:3:> (macex '(macro1))
(do [1 2 3] [1 2 3])
```

I hit this in practice while writing a medium-sized macro, and it took a *lot* of headscratching before I realized that my macro was written correctly, and the presence of a particular value in a separate file was causing it to return the wrong form.

I was able to work around it by using quasiquoting instead:

```
repl:1:> (defmacro macro1 [] (def a ~(1 2 3)) (def b ~[1 2 3]) ~(do ,a ,b))
<function macro1>
repl:2:> (macex '(macro1))
(do (1 2 3) [1 2 3])
```

I don't actually understand why that workaround works, but it let me continue making progress.

**Comment by bakpakin** [comment]
The non-local effects here are certainly weird, and off-putting enough that it might make sense to break the semantics here such that bracketed tuples are no longer equal to normal (parenthesized) tuples.

**Comment by bakpakin** [comment]
technically, the bug is in the compiler here but this is something that is easier to fix by changing the semantics of tuple equality.

**Comment by bakpakin** [comment]
Fixed in aa5c987a94e79455e27094ed427c16ffdb90bb40
